localstack/services/apigateway/integration.py

```diff
@@ -439,134 +427,170 @@ def invoke(self, invocation_context: ApiInvocationContext):
+        # determine target method via reflection
+        dynamo_client = aws_stack.connect_to_service("dynamodb")
+        method_name = camel_to_snake_case(action)
+        client_method = getattr(dynamo_client, method_name, None)
+        if not client_method:
+            raise Exception(f"Unsupported action {action} in API Gateway integration URI {uri}")
 
-            response_template = json.loads(response_template)
-            if response_template["TableName"] != table_name:
-                # TODO: check if this error is valid (AWS parity)
-                msg = "Invalid table name specified in integration response template."
-                return make_error_response(msg, 404)
+        # run request against DynamoDB backend
+        response = client_method(**payload)
 
-            dynamo_client = aws_stack.connect_to_resource("dynamodb")
-            table = dynamo_client.Table(table_name)
+        # apply response templates
+        response_content = json.dumps(remove_attributes(response, ["ResponseMetadata"]))
+        response_obj = requests_response(content=response_content)
+        response = self.response_templates.render(invocation_context, response=response_obj)
 
-            event_data = {}
-            data_dict = json.loads(data)
-            for key, _ in response_template["Item"].items():
-                event_data[key] = data_dict[key]
+        # construct final response
+        response = requests_response(response)
+        invocation_context.response = response
+        return response
 
-            table.put_item(Item=event_data)
-            response = requests_response(event_data)
-            return response
 
-        if "Query" in action:
-            template = integration["requestTemplates"].get(APPLICATION_JSON)
+class S3Integration(BackendIntegration):
+    # target ARN patterns
+    TARGET_REGEX_PATH_S3_URI = (
+        r"^arn:aws:apigateway:[a-zA-Z0-9\-]+:s3:path/(?P<bucket>[^/]+)/(?P<object>.+)$"
+    )
+    TARGET_REGEX_ACTION_S3_URI = r"^arn:aws:apigateway:[a-zA-Z0-9\-]+:s3:action/(?:GetObject&Bucket\=(?P<bucket>[^&]+)&Key\=(?P<object>.+))$"
 
-            if template is None:
-                msg = "No request template is defined in the integration."
-                LOG.info("%s Existing: %s", msg, response_templates)
-                return make_error_response(msg, 404)
+    def invoke(self, invocation_context: ApiInvocationContext):
+        invocation_path = invocation_context.path_with_query_string
+        integration = invocation_context.integration
+        path_params = invocation_context.path_params
+        relative_path, query_string_params = extract_query_string_params(path=invocation_path)
+        uri = integration.get("uri") or integration.get("integrationUri") or ""
 
-            response_template = response_templates.get(APPLICATION_JSON)
+        s3 = aws_stack.connect_to_service("s3")
+        uri = apply_request_parameters(
```

Comment:
in next iterations we can start moving from this function to the request parameters resolver

Reply:
Sounds great, looking forward to it! Do you have this in the pipeline @calvernaz ? Let's coordinate..